src/apps/common/submodules/portWizard/portWizard.js

```diff
@@ -173,25 +224,59 @@ define(function(require) {
 		 * @returns  {Object}  Object that contains the updated step data, and if it is valid
 		 */
 		portWizardNameAndNumbersUtil: function($template, args) {
-			var self = this;
-
-			// TODO: Not implemented
+			var self = this,
+				$form = $template.find('form'),
+				isValid = monster.ui.valid($form),
+				nameAndNumbersData;
+
+			if (isValid) {
+				nameAndNumbersData = monster.ui.getFormData($form.get(0));
+
+				// Extract and format numbers
+				nameAndNumbersData.numbersToPort.formattedNumbers = _
+					.chain(nameAndNumbersData.numbersToPort.numbers)
+					.split(',')
+					.map(_.trim)
+					.reject(_.isEmpty)
+					.map(monster.util.getFormatPhoneNumber)
+					.uniqBy('e164Number')
+					.value();
+
+				nameAndNumbersData.numbersToPort.numbers = _
+					.chain(nameAndNumbersData.numbersToPort.formattedNumbers)
+					.map('originalNumber')
```

Comment:
What do you think of displaying numbers in their user format to have consistent formatting?

Reply:
I would prefer it that way, but I think that some users may get confused if they see any of the numbers in a different format. They may feel that it has been changed. What do you think?
Similar to this, I was also considering to remove the repeated numbers.

Reply:
> I would prefer it that way, but I think that some users may get confused if they see any of the numbers in a different format. They may feel that it has been changed. What do you think?

Those numbers would be rendered using the user preferred format (hopefully) in the next step anyways so we might as well do it here.

My use case was if someone copy pasted number that would be in different format (international, national, e164) it will not look uniform.

Since the only reason a user would come back to this step would be to edit a number, maybe using `e164` would make more sense as it is easier to edit than in national format?

> Similar to this, I was also considering to remove the repeated numbers.

Yes, that would be expected, they can always review which numbers where added in the next step.